src/main/java/frc/robot/subsystems/Vision.java

```diff
@@ -0,0 +1,86 @@
+package frc.robot.subsystems;
+
+import org.photonvision.PhotonCamera;
+import org.photonvision.targeting.PhotonPipelineResult;
+import org.photonvision.targeting.PhotonTrackedTarget;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class Vision extends SubsystemBase
+{
+    private static Vision  _instance;
+    
+    public static Vision getInstance()
+    {
+        if (_instance == null) 
+        {
+            _instance = new Vision();
+        }
+       
+        return _instance;
+    }
+   
+    private final PhotonCamera _camera0 = new PhotonCamera("camera1");
+    private final PhotonCamera _camera1 = new PhotonCamera("camera2");
+    private PhotonPipelineResult _latestResult0 = new PhotonPipelineResult();
+    private PhotonPipelineResult _latestResult1 = new PhotonPipelineResult();
+    
+   
+    private Vision()
+    {
+
+    } 
```

Comment:
Declare the variable and its type above the constructor and then initialize it inside the constructor. We may also want better names for the variables instead of "_camera0".

Reply:
I agree with Joe. In practice, that means the last several lines will look like this:

```java
private final PhotonCamera _camera0;
private final PhotonCamera _camera1;
private PhotonPipelineResult _latestResult0;
private PhotonPipelineResult _latestResult1;


private Vision()
{
    _camera0 = new PhotonCamera("camera1");
    _camera1 = new PhotonCamera("camera2");
    _latestResult0 = new PhotonPipelineResult();
    _latestResult1 = new PhotonPipelineResult();
}
```

The reason we do this is because we want to give everything a chance to turn on before we start trying to communicate with it. The constructor for a `PhotonCamera` likely tries to communicate with the camera, and we'll likely get an error when it tries to connect if the camera hasn't initialized yet. By putting this in the constructor, we're delaying that step just a little bit.